java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessagingFactory.java

```diff
@@ -136,14 +137,17 @@ private void startReactor(ReactorHandler reactorHandler) throws IOException
 	}
 
 	@Override
-	public Connection getConnection()
+	public Session getSession(final String sessionId)
 	{
 		if (this.connection == null || this.connection.getLocalState() == EndpointState.CLOSED || this.connection.getRemoteState() == EndpointState.CLOSED)
 		{
 			this.connection = this.getReactor().connectionToHost(this.hostName, ClientConstants.AMQPS_PORT, this.connectionHandler);
 		}
 
-		return this.connection;
+		final Session session = this.connection.session();
+		session.open();
+		
+		return session;
```

Comment:
>session [](start = 9, length = 7)

so does it mean we will potentially share the same session for multiple links going forward? Right now we have one factory per sender/receiver but if this assumption change (for example for messaging API) then will this be a problem? (I think today in .Net we still have session:link as a 1:1 concept.